Source/fontconfig/FCFontEnumerator.m

```diff
@@ -53,6 +53,87 @@
 #define FC_WEIGHT_ULTRABLACK FC_WEIGHT_BLACK
 #endif
 
+static float
+convertWeight (int fcWeight, int bottomValue, int topValue)
+{
+  /*
+    This is the distance between topValue and bottomValue expressed as a
+    fraction between zero and one. We do this to express the range of
+    fontconfig font weights in a useful manner.
+  */
+
+  if (fcWeight <= bottomValue)
+    {
+      return 0.0;
+    }
+  else if (fcWeight >= topValue)
+    {
+      return 1.0;
+    }
+  else
+    {
+      return (float) (fcWeight - bottomValue) * (1.0f / (topValue - bottomValue));
+    }
+}
+
+static NSComparisonResult
+sortFontFacesArray(id fontArr1, id fontArr2, void *context)
+{
+  /*
+    Order of array:
```

Comment:
This comment looks incorrect to me when compared to the following code.

Reply:
Which comment? The convertWeight() function does do what the comment says it does, convert a number into a fraction accurately representing the numeric position of "fcWeight" within the range between "bottomValue" and "topValue", while enforcing the rule that fcWeight must indeed be within that range. It's for assigning fractional weight values for fonts that do not match any of the pre-defined Fontconfig weight classes -- which turns out to not be a rare thing, after seeing GNUstep assign "medium" weights to fonts that Fontconfig knew were Ultra Light.

The git comment attached to the commit describes what's happening in detail.

Reply:
The comment I was referring to is the one about the order of the array. This comment states a certain order and the sorting function is enforcing a different one. One of the two sides is incorrect and I would like to have them consistent.